wagtailmenus/models/menuitems.py

```diff
@@ -96,6 +96,16 @@ def relative_url(self, site=None):
                 return ''
         return self.link_url + self.url_append
 
+    def get_full_url(self):
+        if self.link_page:
+            try:
+                return self.link_page.full_url + self.url_append
```

Comment:
I feel as though this method would benefit from accepting a `request` parameter, and passing that to the new `link_page.get_full_url()` method if the page has it.